### PR TITLE
Expand globs eagerly; Add arguments to `start` & `watch`

### LINKED ISSUE #178: Eager glob expansion
Currently the code calls `expand` aka `globby` each time `unwrap` is called. It seems like it would be beneficial to eagerly evaluate in `source`, so it is done once for each `source`, rather than multiple times per wrap/target.

### LINKED ISSUE #187: Ability to pass in arguments to `start` and `watch` tasks
It would be nice if you could pass in arguments to tasks when they're called from `fly.start` and `fly.watch`

@bucaran Here's an example of how I think it could work

``` js
const paths = {
  styles: 'app/lib/styles/**/*',
  js: 'app/lib/js/**/*'
}

export default async function() {
  this.start([
    // yes it's the same as the `styles` task but it's an example.
    // the key value in the object would be the name of the task
    { styles: paths.styles }, 
    // you could still pass in a string for backwards compatibility
    'js'
  ], { parallel: true }) // still pass in options as normal
}

export async function styles(globs) {
  await this
    .source(globs || paths.styles)
    .scss()
    .target(dist)
}

export async function js(globs) {
  await this
    .source(globs || paths.js)
    .babel()
    .target(dist)
}

export async function watch() {
  // `fly.watch` would update to be able to accept a function.
  // This would just be an addition to accepting an `array` not replacing it,
  // this way the changes can be backwards compatible
  const styles = this.watch(paths.styles, async (changed) => {
    // the initial time this function runs the `changed` variable could be
    // the glob that was passed (aka `paths.styles`), then after that `changed` 
    // would be the files that changed. This way users have the option to pass in 
    // just the files that changed, which would save a ton of time on tasks
    await this.start({ styles: changed })
  })
  const js = this.watch(paths.js, async (changed) => {
    await this.start({ js: changed })
  })

  // other watch tasks...

  await Promise.all([ styles, js/*, ...other tasks */ ])
}
```

The reason why I bring this addition up is because there's several times I've had tasks that are looking at all `js` files for changes and the js task is clearing the **entire**  js dist directory and then rebuilding it. That becomes quite time consuming when I'm just making a simple change to 1 file and all I really need is to run that 1 file through the task.

### PR DESCRIPTION
**Pull request by tjbenton**
This pull request solves #178 and #187

### CODE REVIEW

**Comment by tjbenton** [comment]
Not sure why the test is failing on `0.12.15` I switch to use `0.12.15` and reinstalled everything and it's working just fine for me.

screen shot 2016-08-15 at 12 13 36 pm

File: lib/fly.js [hunk 1]
```diff
@@ -81,16 +81,28 @@ module.exports = Fly
...
- * @param  {String|Array} globs The glob selection
+ * @param  {String|Array} ...globs The glob selection
+ * @param {Object} options If the last argument is an object it's used as options to pass `node-glob`
  * @return                      The current Fly instance.
  */
-Fly.prototype.source = function (globs) {
-	globs = arrify(globs)
+Fly.prototype.source = function () {
+	var globs = flatten([].slice.call(arguments))
+	var options = globs.pop()
```
Comment (lukeed) [inline lib/fly.js]: If the last option **isn't** an option object then you're losing a `glob` entry for the `map` function below.

File: lib/fly.js [hunk 2]
```diff
@@ -81,16 +81,28 @@ module.exports = Fly
...
+
+	if (Object.prototype.toString.call(options) !== '[object Object]') {
```
Comment (lukeed) [inline lib/fly.js]: How about `options !== Object(options)` or `typef options !== 'object` ?

**Comment by lukeed** [comment]
This looks great @tjbenton, thank you! 😄 

I reran the Travis test & it's passing now. 💯 

Also, do you mind fixing the object check I added & adding a test for the new `start` options?

Thanks!

File: lib/fly.js (hunk 2 above)
Comment (tjbenton) [inline lib/fly.js]: The to string method that I put in there is the only reliable one. If you run this test in the console you should see that what I'm talking about.

``` js
var types = {
  number: 1,
  array: [],
  string: '',
  object: {},
  boolean: true
}

for (var key in types) {
  console.log(key)
  var type = types[key]
  console.log("typeof " + key + " !== 'object' = ", typeof type !== 'object')
  console.log(key + ' !== Object(array) =', type !== Object(array))
  console.log('to string method:', Object.prototype.toString.call(type) !== '[object Object]')
  console.log('')
}
```

if `options` is an array and you try to use either of those methods it's going to return `false` when it should return `true`. 

[Here's](http://stackoverflow.com/questions/5876332/how-can-i-differentiate-between-an-object-literal-other-javascript-objects) an issue on stackoverflow about type checking for a plain object

Comment (lukeed) [inline lib/fly.js]: ok 👌 thanks for that

**Comment by ghost** [comment]
Thanks so much @tjbenton!

**Comment by lukeed** [comment]
As soon as a test (or two) is added for the new behavior, I'll get this in!

**Comment by tjbenton** [comment]
@bucaran @lukeed Glad I could help out

**Pull request merged**